### ISSUE TITLE
ModRamFunctions needs legacy control structure updating

### ISSUE THREAD

**Issue by drsteve** [lanl/RAM-SCB#97]
While we can't necessarily control some of our dependencies, there are still a couple of GOTO statements in the codebase (`ModRamFunctions.f90`). These are coupled with `DO [line number] ... CONTINUE` loop control, and can be rewritten to conform to modern Fortran standards.

These are an ideal case for some unit tests. So first we can write unit tests for functions with old-style control flow (and make sure that the tests are thorough and pass), then update the functions to use modern approaches.

**Comment by drsteve** [comment]
Having written some tests, it looks like `ERFF` is just the error function which, as of Fortran2008, is an elemental function. The functions that use the old-style loops and gotos (`GSER`, `GCF`, `GAMMLN`) are all _only_ called when using `ERFF`. The same is true for `GAMMP`, although that is modern-looking fortran.

So I think that, rather than testing all these functions and modernizing, we just need to test `Gcoul` and replace the calls to `ERFF` (in `Gcoul`, and in a couple of places in `ModRamCoul`) with calls to the elemental function `erf`.

The question is... are we okay changing code to include a feature that is only part of the Fortran standard starting with 2008?
Any thoughts @Pheosics ?

**Comment by drsteve** [comment]
Similarly, I'd like to update the function definitions so instead of
```
function foo(x)
    y = [code]
    return
end function foo
```
we are explicit (new-style), like
```
function foo(x) result(y)
    y = [code]
end function foo
```

I should probably make a new ticket for that, but happy to take comments if there's a downside to updating all that.

**Comment by Pheosics** [comment]
I see no issues updating the code to be compliant with a 14 year old standard.

But in all seriousness, the entire ModRamFunctions section should be reevaluated as most of those are copy paste from numerical recipes handbook, and I am really not sure that we need them anymore.

As for the explicit function return, I see no reason not to. Functions in fortran are already weird to me.

**Comment by drsteve** [comment]
I'll do the `erf` replacement once #96 is merged in, so I can do it as two commits:
1. Change `erff` calls to use intrinsic `erf`
2. Remove test for `erff`

That way the tests will pass after commit 1, showing that the change hasn't affected results.

I'll do code modernization stuff later.
